client/src/pages/SignUp.js

```diff
@@ -78,11 +79,14 @@ const SignUpPage = (props) => {
   const [passwordError, setPasswordError] = useState("");
   const [passwordConfirmError, setPasswordConfirmError] = useState("");
   const [message, setMessage] = useState(null);
+  const [location, setLocation] = useState('')
 
-
+  const locationState = {location,setLocation}
+  
   const handleSignUp = (e) => {
     e.preventDefault();
-    const data = {email: email, password: password, name: username};
+    const data = {email: email, password: password, name: username, location};
```

Comment:
since the rest of the object is explicitly defined, you can do `location: location` to be consistent.
Probably the best option is to do `const data = {email, password, username, location};` and change the property from `name` to `username` on the backend since that's confusing naming anyways.